airbyte-integrations/connectors/source-declarative-manifest/source_declarative_manifest/run.py

```diff
@@ -11,9 +11,47 @@
 from airbyte_cdk.entrypoint import AirbyteEntrypoint, launch
 from airbyte_cdk.models import AirbyteMessage, ConnectorSpecification, Type
 from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
+from airbyte_cdk.sources.declarative.yaml_declarative_source import YamlDeclarativeSource
+
+
+class SourceLocalYaml(YamlDeclarativeSource):
+    """
+    Declarative source defined by a yaml file in the local filesystem
+    """
+
+    def __init__(self):
+        """
+        HACK!
+            Problem: YamlDeclarativeSource relies on the calling module name/path to find the yaml file.
+            Implication: If you call YamlDeclarativeSource directly it will look for the yaml file in the wrong place. (e.g. the airbyte-cdk package)
+            Solution: Subclass YamlDeclarativeSource from the same location as the manifest to load.
+
+            When can we remove this?
+                When the airbyte-cdk is updated to not rely on the calling module name/path to find the yaml file.
+                When all manifest connectors are updated to use the new airbyte-cdk.
+                When all manifest connectors are updated to use the source-declarative-manifest as the base image.
+        """
+        super().__init__(**{"path_to_yaml": "manifest.yaml"})
+
+
+def _is_local_manifest_command(args: List[str]) -> bool:
+    # Check for a local manifest.yaml file
+    return os.path.isfile("source_declarative_manifest/manifest.yaml")
```

Comment:
I'm sure I'm missing something here, but the `source_declarative_manifest/` part of this path would be different for different connectors, right? 

Are we hard-coding that path somewhere and copying the manifest.yaml file into it?

Reply:
Ah, is that here? https://github.com/airbytehq/airbyte/pull/39906/files#diff-d6e16bca0698a2df60ee548c530e7a517841d23743762e6137f82c12e70e1a26R54

Reply:
Yup that is the place.

The reason we do it this way is due to the cdk using 
```py
        yaml_config = pkgutil.get_data(package, path_to_yaml_file)
```
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/yaml_declarative_source.py#L27

Which requires the file to be part of the calling package.

We could have changed the CDK, but I dont want manifest only connectors to depend on the most recent version of the CDK and potentially block the migration of an old connector using an old cdk version.